src/main/java/appeng/client/render/cablebus/CableBusRenderState.java

```diff
@@ -153,4 +154,43 @@ public void setCableBusAdjacent( EnumSet<EnumFacing> cableBusAdjacent )
 		return this.boundingBoxes;
 	}
 
+	@Override
+	public int hashCode()
+	{
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ( ( this.attachmentConnections == null ) ? 0 : this.attachmentConnections.hashCode() );
+		result = prime * result + ( ( this.cableBusAdjacent == null ) ? 0 : this.cableBusAdjacent.hashCode() );
+		result = prime * result + ( ( this.cableColor == null ) ? 0 : this.cableColor.hashCode() );
+		result = prime * result + ( ( this.cableType == null ) ? 0 : this.cableType.hashCode() );
+		result = prime * result + ( ( this.channelsOnSide == null ) ? 0 : this.channelsOnSide.hashCode() );
+		result = prime * result + ( ( this.connectionTypes == null ) ? 0 : this.connectionTypes.hashCode() );
+		result = prime * result + ( ( this.coreType == null ) ? 0 : this.coreType.hashCode() );
+		return result;
+	}
+
+	@Override
+	public boolean equals( Object obj )
+	{
+		if( this == obj )
+		{
+			return true;
+		}
+		if( obj == null )
+		{
+			return false;
+		}
+		if( getClass() != obj.getClass() )
+		{
+			return false;
+		}
+
+		final CableBusRenderState other = (CableBusRenderState) obj;
+
+		return this.cableColor == other.cableColor && this.cableType == other.cableType && this.coreType == other.coreType && Objects
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (6) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1067)